File: BackEnd (Java)/UtilsArmyModel/src/main/java/pe/edu/pucp/utilsarmy/gestion_de_material/model/MaterialBibliografico.java
package pe.edu.pucp.utilsarmy.gestion_de_material.model;

import java.util.ArrayList;
import java.util.stream.Collectors;

public class MaterialBibliografico {

    private int idMaterial;
    private String titulo;
    private int anho_publicacion;
    private int numero_paginas;
    private EstadoMaterial estado;
    private String clasificacion_tematica;
    private boolean activo;
    private String idioma;
    private String editoriales;
    private ArrayList<Contribuyente> contribuyentes;
    private ArrayList<Ejemplar> ejemplares;
    private ArrayList<Biblioteca> bibliotecas;
    private String autoresTexto;
    private String bibliotecasTexto;
    private TipoMaterial tipo;
    private int cantidadDisponible;
    public MaterialBibliografico() {
        this.contribuyentes = new ArrayList<>();
        this.ejemplares = new ArrayList<>();
        this.bibliotecas = new ArrayList<>();
    }

    public MaterialBibliografico(String titulo,
            int anho_publicacion, int numero_paginas, EstadoMaterial estado,
            String clasificacion_tematica, boolean activo, String idioma, String editoriales) {
       
        this.titulo = titulo;
        this.anho_publicacion = anho_publicacion;
        this.numero_paginas = numero_paginas;
        this.estado = estado;
        this.clasificacion_tematica = clasificacion_tematica;
        this.activo = activo;
        this.idioma = idioma;
        this.contribuyentes = new ArrayList<>();
        this.ejemplares = new ArrayList<>();
        this.editoriales = editoriales;
    }

    public int getIdMaterial() {
        return idMaterial;
    }

    public void setIdMaterial(int idMaterial) {
        this.idMaterial = idMaterial;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public int getAnho_publicacion() {
        return anho_publicacion;
    }

    public void setAnho_publicacion(int anho_publicacion) {
        this.anho_publicacion = anho_publicacion;
    }

    public int getNumero_paginas() {
        return numero_paginas;
    }

    public void setNumero_paginas(int numero_paginas) {
        this.numero_paginas = numero_paginas;
    }

    public EstadoMaterial getEstado() {
        return estado;
    }

    public void setEstado(EstadoMaterial estado) {
        this.estado = estado;
    }

    public String getClasificacion_tematica() {
        return clasificacion_tematica;
    }

    public void setClasificacion_tematica(String clasificacion_tematica) {
        this.clasificacion_tematica = clasificacion_tematica;
    }

    public boolean isActivo() {
        return activo;
    }

    public void setActivo(boolean activo) {
        this.activo = activo;
    }

    public String getIdioma() {
        return idioma;
    }

    public void setIdioma(String idioma) {
        this.idioma = idioma;
    }

    public ArrayList<Contribuyente> getContribuyentes() {
        return contribuyentes;
    }

    public void setContribuyentes(ArrayList<Contribuyente> contribuyentes) {
        this.contribuyentes = contribuyentes;
    }

    public ArrayList<Ejemplar> getEjemplares() {
        return ejemplares;
    }

    public void setEjemplares(ArrayList<Ejemplar> ejemplares) {
        this.ejemplares = ejemplares;
    }

    public String getAutoresTexto() {
        return autoresTexto;
    }

    public void setAutoresTexto(String autoresTexto) {
        this.autoresTexto = autoresTexto;
    }

    public void FormatoAutores() {
        if (contribuyentes == null || contribuyentes.isEmpty()) {
            autoresTexto = "No registrados";
        } else {
            autoresTexto = contribuyentes.stream()
                    .map(a -> a.getNombre() + " " + a.getPrimer_apellido())
                    .collect(Collectors.joining(", "));
        }

    }
    public void FormatoBibliotecas() {
        if (bibliotecas == null || bibliotecas.isEmpty()) {
            bibliotecasTexto = "No hay bibliotecas con este material";
        } else {
            bibliotecasTexto = bibliotecas.stream()
                    .map(a -> a.getNombre())
                    .collect(Collectors.joining(", "));
        }

    }

    public TipoMaterial getTipo() {
        return tipo;
    }

    public void setTipo(TipoMaterial tipo) {
        this.tipo = tipo;
    }

    public int getCantidadDisponible() {
        return cantidadDisponible;
    }

    public void setCantidadDisponible(int cantidadDisponible) {
        this.cantidadDisponible = cantidadDisponible;
    }

    public ArrayList<Biblioteca> getBibliotecas() {
        return bibliotecas;
    }

    public void setBibliotecas(ArrayList<Biblioteca> bibliotecas) {
        this.bibliotecas = bibliotecas;
    }

    public String getBibliotecasTexto() {
        return bibliotecasTexto;
    }

    public void setBibliotecasTexto(String bibliotecasTexto) {
        this.bibliotecasTexto = bibliotecasTexto;
    }

    public String getEditoriales() {
        return editoriales;
    }

    public void setEditoriales(String editoriales) {
        this.editoriales = editoriales;
    }

}

File: FrontEnd (C#)/SoftProgModel/GestMaterial/MaterialBibliografico.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgModel.GestMaterial
{

    public class MaterialBibliografico
    {
        private int idMaterial;
        private string titulo;
        private int anho_publicacion;
        private int numero_paginas;
        private EstadoMaterial estado;
        private string clasificacion_tematica;
        private bool activo;
        private string idioma;
        private TipoMaterial tipo;
        private BindingList<Editorial> editoriales;
        private BindingList<Contribuyente> contribuyentes;
        private BindingList<Ejemplar> ejemplares;
        private BindingList<Biblioteca> bibliotecas;
        private int cantidadDisponibles;
        public string AutoresTexto
        {
            get
            {
                if (contribuyentes == null || contribuyentes.Count == 0)
                    return "No registrados";
                return string.Join(", ", contribuyentes.Select(a => a.Nombre + " " + a.Primer_apellido));
            }
        }
        public string BibliotecasTexto
        {
            get
            {
                if (bibliotecas == null || bibliotecas.Count == 0)
                    return "No disponible";
                return string.Join(", ", bibliotecas.Select(a => a.Nombre));
            }
        }

        public BindingList<Editorial> getEditoriales()
        {
            return Editoriales;
        }

        public void setEditoriales(BindingList<Editorial> editoriales)
        {
            this.Editoriales = editoriales;
        }

        public MaterialBibliografico()
        {
            this.Editoriales = new BindingList<Editorial>();
            this.Contribuyentes = new BindingList<Contribuyente>();
            this.Ejemplares = new BindingList<Ejemplar>();
            this.bibliotecas = new BindingList<Biblioteca>();
        }

        public MaterialBibliografico(int idMaterial, string titulo,
                int anho_publicacion, int numero_paginas, EstadoMaterial estado,
                string clasificacion_tematica, bool activo, string idioma, TipoMaterial tipo)
        {
            this.IdMaterial = idMaterial;
            this.Titulo = titulo;
            this.Anho_publicacion = anho_publicacion;
            this.Numero_paginas = numero_paginas;
            this.Estado = estado;
            this.Clasificacion_tematica = clasificacion_tematica;
            this.Activo = activo;
            this.Idioma = idioma;
            this.Tipo = tipo;
            this.Contribuyentes = new BindingList<Contribuyente>();
            this.Ejemplares = new BindingList<Ejemplar>();
        }



        public int IdMaterial { get => idMaterial; set => idMaterial = value; }
        public string Titulo { get => titulo; set => titulo = value; }
        public int Anho_publicacion { get => anho_publicacion; set => anho_publicacion = value; }
        public int Numero_paginas { get => numero_paginas; set => numero_paginas = value; }
        public EstadoMaterial Estado { get => estado; set => estado = value; }
        public string Clasificacion_tematica { get => clasificacion_tematica; set => clasificacion_tematica = value; }
        public bool Activo { get => activo; set => activo = value; }
        public string Idioma { get => idioma; set => idioma = value; }
        public TipoMaterial Tipo { get => tipo; set => tipo = value; }
        public BindingList<Editorial> Editoriales { get => editoriales; set => editoriales = value; }
        public BindingList<Contribuyente> Contribuyentes { get => contribuyentes; set => contribuyentes = value; }
        public BindingList<Ejemplar> Ejemplares { get => ejemplares; set => ejemplares = value; }
        public BindingList<Biblioteca> Bibliotecas { get => bibliotecas; set => bibliotecas = value; }
        public int CantidadDisponibles { get => cantidadDisponibles; set => cantidadDisponibles = value; }
    }
}
